enhancements/network/ovn-kubernetes-localnet-api.md

```diff
@@ -0,0 +1,557 @@
+knowledge that require cluster-admin intervention.
+Such as configuring the OVS switch to which the localnet network connects to. 
+
+Managing localnet topology networks using a well-formed API using the CUDN CRD could improve UX as it is managed by a 
+controller, perform validations and reflect the state via status.
+
+### User Stories
+
+#### Definition of personas:
+Admin - is the cluster admin.
+User - non cluster-admin user, project manager.
+Workloads - pod or VMs (KubeVirt).
+
+- As an admin I want to create a user-defined network over localnet topology using CUDN CRD. 
+  - In case the network configuration is bad I want to get an informative message saying what went wrong.
+- As an admin I want to enable users to connect workloads in project/namespaces they have permission to, to the localnet network I created for them.
+- As a user I want to be able to connect my workloads (pod/VMs) to the localnet the admin created in my namespace.
+- As a user I want my workloads to be able to communicate with each other over the localnet network.
+- As a user I want my connected VMs to the localnet network to be able to migrate from one node to another, having its localnet network interface IP address unchanged.
+
+### Goals
+
+- Enable creating user-defined-networks over localnet topology using OVN-K CUDN CRD.
+- Streamline localnet UX: detect misconfigurations early, communicate issues via informative status conditions and or events.
+
+### Non-Goals
+
+- This proposal does not cover changes for network configurations on day2 (mutable network configuration).
+
+## Proposal
+
+### Summary
+Extend the CUDN CRD to enable creating user-defined networks over localnet topology.
+Since the CUDN CRD is targeted for cluster-admin users, it enables preventing non-admin users performing changes that 
+could disrupt the cluster or impact the physical network to which the workloads would connect to.
+
+### Localnet using NetworkAttachmentDefinition
+As of today OVN-K enables multi-homing including localnet topology networks using NADs
+The following NAD YAML describe localnet topology configuration and options:
+
+```yaml
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenantblue
+  namespace: blue
+spec:
+    config: > '{
+        "cniVersion": "0.3.1",
+        "type": "ovn-k8s-cni-overlay"
+        "netAttachDefName": "blue/tenantblue",
+        "topology": "localnet",
+        "name": "tenantblue",#1
+        "physicalNetworkName": "mylocalnet1", #2
+        "subnets": "10.100.0.0/24", #3
+        "excludeSubnets": "10.100.50.0/32", #4
+        "vlanID": "200", #5
+        "mtu": "1500", #6
+        "allowPersistentIPs": true #7
+    }'
+```
+1. `name` 
+   The underlying network name.
+   - Should match the node OVS bridge-mapping network-name.
+   - In case Kubernetes-nmstate is used, should match the `NodeNetworkConfigurationPolicy` (NNCP) `spec.desiredState.ovn.bridge-mappings` value.
```

Comment:
should actually match the `localnet` attribute of an item on that slice / array.

Reply:
Added text to clarify that